### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by vstrizhakov**
### Description of the Change

- Fixed issue #288 
- Some refactoring was done
- Created migration for removed old table bot_chat_bot_user
- Prepared migrations for HR module
- Added FillablePropertiesController for using it in cases when we need to collect an array of data when filling some object such as Company, Vacancy, Resume, etc. For now it supports collecting data without it's validation - this will be done in future commits.

### CODE REVIEW

File: modules/bot/controllers/privates/RandomIntController.php [hunk 1]
```diff
@@ -14,18 +14,17 @@
...
-        return [
-            new SendMessageCommand(
-                $this->getTelegramChat()->chat_id,
-                $randomInt
-            ),
-        ];
+        return ResponseBuilder::fromUpdate($this->getUpdate())
+            ->sendMessage(
+                new MessageText($randomInt),
+            )
```
[the remark this answers is not in the document]
Comment (vstrizhakov) [reply modules/bot/controllers/privates/RandomIntController.php]: Solved

File: modules/bot/components/response/ResponseBuilder.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+     * @param array|null $replyMarkup
+     * @return $this
+     */
+    public function editMessageReplyMarkup(
+        array $replyMarkup = null
+    )
+    {
+        if ($callbackQuery = $this->update->getCallbackQuery()) {
+            $this->answerCallbackQuery();
+            $$this->commands[] = new EditMessageReplyMarkupCommand(
```
[the remark this answers is not in the document]
Comment (vstrizhakov) [reply modules/bot/components/response/ResponseBuilder.php]: Solved

File: modules/bot/components/response/ResponseBuilder.php [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        return $this;
+    }
+
+    /**
+     * @param MessageText $messageText
+     * @param array|null $replyMarkup
+     * @param bool $disablePreview
+     * @return $this
+     */
+    public function sendMessage(MessageText $messageText, array $replyMarkup = null, bool $disablePreview = false)
```
[the remark this answers is not in the document]
Comment (vstrizhakov) [reply modules/bot/components/response/ResponseBuilder.php]: This parameter is passed to the constructor of SendMessage command, it does not have a logical value

File: migrations/m200315_102720_create_vacancy_table.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%vacancy}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'company_id' => $this->integer()->unsigned()->notNull(),
+            'views' => $this->integer()->notNull()->defaultValue(0),
+            'status' => $this->integer()->notNull()->defaultValue(0),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: tinyInteger()

File: modules/bot/views/privates/companies/show.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<b><?= \Yii::t('bot', 'Name of company: ')?></b> <?= $name ?>
+<br/>
+<b><?= \Yii::t('bot', 'Address: ')?></b> <?= $address ?>
```
Comment (grandmotivator) [inline modules/bot/views/privates/companies/show.php]: <b><?= \Yii::t('bot', 'Address')?></b>: <?= $address ?>

File: migrations/m200217_224106_create_company_user_table.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%company_user}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'user_id' => $this->integer()->unsigned()->notNull(),
+            'company_id' => $this->integer()->unsigned()->notNull(),
+            'user_role' => $this->integer()->unsigned()->notNull()->defaultValue(0),
```
Comment (grandmotivator) [inline migrations/m200217_224106_create_company_user_table.php]: tinyInteger()

File: migrations/m200315_102720_create_vacancy_table.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function safeUp()
+    {
+        $this->createTable('{{%vacancy}}', [
+            'id' => $this->primaryKey()->unsigned(),
+            'company_id' => $this->integer()->unsigned()->notNull(),
+            'views' => $this->tinyInteger()->notNull()->defaultValue(0),
+            'status' => $this->tineInteger()->notNull()->defaultValue(0),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: tinyInteger()

File: migrations/m200315_102720_create_vacancy_table.php [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+            'hours_of_employment' => $this->string()->notNull(),
+            'requirements' => $this->text()->notNull(),
+            'salary' => $this->string()->notNull(),
+            'skills_description' => $this->text()->notNull(),
+            'conditions' => $this->text()->notNull(),
+            'responsibility' => $this->text()->notNull(),
+            'sex' => $this->integer()->notNull(),
+            'location_lat' => $this->integer(),
+            'location_lon' => $this->integer(),
+            'location_at' => $this->integer(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: unsigned()

File: migrations/m200315_102720_create_vacancy_table.php [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            'name' => $this->string()->notNull(),
+            'employment' => $this->string()->notNull(),
+            'hours_of_employment' => $this->string()->notNull(),
+            'requirements' => $this->text()->notNull(),
+            'salary' => $this->string()->notNull(),
+            'skills_description' => $this->text()->notNull(),
+            'conditions' => $this->text()->notNull(),
+            'responsibility' => $this->text()->notNull(),
+            'sex' => $this->integer()->notNull(),
+            'location_lat' => $this->integer(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: string(255)

File: migrations/m200315_102720_create_vacancy_table.php [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+            'location_lon' => $this->integer(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: string(255)

File: migrations/m200315_102720_create_vacancy_table.php [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            'location_at' => $this->integer(),
+            'created_at' => $this->integer()->notNull(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: unsigned()

File: migrations/m200315_102720_create_vacancy_table.php [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            'updated_at' => $this->integer()->notNull(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: unsigned()

File: migrations/m200315_102720_create_vacancy_table.php [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+            'renewed_at' => $this->integer()->notNull(),
```
Comment (grandmotivator) [inline migrations/m200315_102720_create_vacancy_table.php]: unsigned()

File: migrations/m200315_102720_create_vacancy_table.php (hunk 7 above)
Comment (grandmotivator) [reply migrations/m200315_102720_create_vacancy_table.php]: unsigned()

**Pull request merged**